application/apps/indexer/processor/src/map.rs

```diff
@@ -196,20 +239,44 @@ impl SearchMap {
         }
     }
 
-    pub fn set(&mut self, matches: Option<Vec<FilterMatch>>) {
+    pub fn set(&mut self, matches: Option<Vec<FilterMatch>>, mut stats: Option<FiltersStats>) {
         if let Some(matches) = matches {
             self.matches = matches;
         } else {
             self.matches = vec![];
         }
+        if let Some(stats) = stats.take() {
+            self.stats = stats;
+        } else {
+            self.stats = FiltersStats::default();
+        }
```

Comment:
but here we should cover 2 options:
- if stats isn't none - overwrite it
- if none - drop to defaults
checking for is_none covers just the second use case

Reply:
probably it will be suitable:
```
        self.matches = matches.map_or(vec![], |m| { m });
        self.stats = stats.map_or(FiltersStats::default(), |s| { s });
        ```